File: batch/src/main/java/com/isystk/sample/batch/jobs/BaseItemWriter.java
package com.isystk.sample.batch.jobs;

import com.isystk.sample.batch.context.BatchContext;
import com.isystk.sample.batch.context.BatchContextHolder;
import java.util.List;
import lombok.val;
import org.springframework.batch.item.ItemWriter;

public abstract class BaseItemWriter<T> implements ItemWriter<T> {

  @SuppressWarnings("unchecked")
  @Override
  public void write(List<? extends T> items) throws Exception {
    // コンテキストを取り出す
    BatchContext context = BatchContextHolder.getContext();

    // 書き込む
    doWrite(context, (List<T>) items);
  }

  /**
   * 引数に渡されたアイテムを書き込みます。
   *
   * @param context
   * @param items
   */
  protected abstract void doWrite(BatchContext context, List<T> items);
}

File: batch/src/main/kotlin/com/isystk/sample/batch/jobs/BaseItemWriter.kt
package com.isystk.sample.batch.jobs

import com.isystk.sample.batch.context.BatchContext
import com.isystk.sample.batch.context.BatchContextHolder
import org.springframework.batch.item.ItemWriter

abstract class BaseItemWriter<T> : ItemWriter<T> {
    @Throws(Exception::class)
    override fun write(items: List<T>) {
        // コンテキストを取り出す
        val context = BatchContextHolder.context

        // 書き込む
        doWrite(context, items)
    }

    /**
     * 引数に渡されたアイテムを書き込みます。
     *
     * @param context
     * @param items
     */
    protected abstract fun doWrite(context: BatchContext?, items: List<T>?)
}